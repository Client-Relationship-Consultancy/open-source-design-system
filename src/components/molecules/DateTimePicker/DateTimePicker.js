import React from "react"
import DatePicker from "react-datepicker"
import "react-datepicker/dist/react-datepicker.css"

import PropTypes from "prop-types"
import styled from "styled-components"

const Container = styled.div`
  * input {
    width: 180px;
    font-size: 13px;
  }
`
Container.displayName = "DateTimePickerContainer"

const DateTimePicker = props => {
  const { selectedDate, onChange, showTimeSelect, minDate, maxDate, ...others } = props

  return (
    <Container>
      <DatePicker
        selected={selectedDate}
        onChange={onChange}
        showTimeSelect={showTimeSelect}
        showYearDropdown
        timeFormat="HH:mm"
        timeIntervals={15}
        dateFormat="do MMM yyyy HH:mm"
        timeCaption="time"
        minDate={new Date(minDate)}
        maxDate={new Date(maxDate)}
        {...others}
      />
    </Container>
  )
}

DateTimePicker.displayName = "DateTimePicker"

DateTimePicker.defaultProps = {
  showTimeSelect: true,
}

DateTimePicker.propTypes = {
  selectedDate: PropTypes.instanceOf(Date),
  onChange: PropTypes.func,
  showTimeSelect: PropTypes.bool,
  minDate: PropTypes.string,
  maxDate: PropTypes.string,
}

export default DateTimePicker
