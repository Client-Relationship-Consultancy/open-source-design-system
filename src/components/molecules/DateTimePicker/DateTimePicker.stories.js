import React from "react"
import { storiesOf } from "@storybook/react"

import { DateTimePicker } from "index"

const stories = storiesOf("DateTimePicker", module)

class ExampleDateTimePicker extends React.PureComponent {
  state = {
    selectedDate: new Date(),
  }

  onChangeDate = newDate => {
    this.setState({ selectedDate: newDate })
  }

  render() {
    return <DateTimePicker selectedDate={this.state.selectedDate} onChange={this.onChangeDate} />
  }
}

ExampleDateTimePicker.displayName = "ExampleDatePicker"

stories.add("DateTimePicker", () => <ExampleDateTimePicker />)
