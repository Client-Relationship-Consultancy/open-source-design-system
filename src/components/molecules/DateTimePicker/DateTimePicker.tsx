import React from "react";
import DatePicker, { ReactDatePickerProps } from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

import styled from "styled-components";

const Container = styled.div`
  * input {
    width: 180px;
    font-size: 13px;
  }
`;
Container.displayName = "DateTimePickerContainer";

export interface IDateTimePicker extends Omit<ReactDatePickerProps, "maxDate" | "minDate"> {
  selectedDate?: Date | null;
  minDate: string | number | Date;
  maxDate: string | number | Date;
}
export const DateTimePicker: React.FunctionComponent<IDateTimePicker> = props => {
  const { selectedDate, onChange, showTimeSelect, minDate, maxDate, ...others } = props;

  return (
    <Container className="DateTimePicker">
      <DatePicker
        selected={selectedDate}
        onChange={onChange}
        showTimeSelect={showTimeSelect}
        showYearDropdown
        timeFormat="HH:mm"
        timeIntervals={15}
        dateFormat="do MMM yyyy HH:mm"
        timeCaption="time"
        minDate={new Date(minDate)}
        maxDate={new Date(maxDate)}
        {...others}
      />
    </Container>
  );
};

DateTimePicker.displayName = "DateTimePicker";

DateTimePicker.defaultProps = {
  showTimeSelect: true,
};
