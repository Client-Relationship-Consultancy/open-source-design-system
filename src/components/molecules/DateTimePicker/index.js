export { DateTimePicker, IDateTimePicker } from "./DateTimePicker"
