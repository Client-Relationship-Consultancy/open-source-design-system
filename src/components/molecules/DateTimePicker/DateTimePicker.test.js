import React from "react"
import DatePicker from "react-datepicker"
import { shallow } from "enzyme"

import { DateTimePicker } from "./DateTimePicker"

const mockOnChange = jest.fn()

describe("DateTimePicker Component Testing", () => {
  const component = shallow(
    <DateTimePicker
      selectedDate={new Date("12/12/1995 12:45")}
      onChange={mockOnChange}
      minDate="12/11/1995 12:45"
      maxDate="01/01/1996 12:45"
    />,
  )
  it("should have selected date the selectedDate prop", () => {
    expect(component.find(DatePicker).prop("selected")).toEqual(new Date("12/12/1995 12:45"))
  })
  it("should call onChange when DatePicker onChange is triggered", () => {
    component
      .find(DatePicker)
      .props()
      .onChange()

    expect(mockOnChange).toHaveBeenCalled()
  })

  it("should set up datePicker without max and min date", () => {
    const componentWithoutBounds = shallow(
      <DateTimePicker selectedDate={new Date("12/12/1995 12:45")} onChange={mockOnChange} />,
    )
    expect(componentWithoutBounds).toMatchSnapshot()
  })

  it("should match latest snapshot", () => {
    expect(component).toMatchSnapshot()
  })
})
