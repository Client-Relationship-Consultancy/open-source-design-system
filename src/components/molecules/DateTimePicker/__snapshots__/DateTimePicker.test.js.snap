// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker Component Testing should match latest snapshot 1`] = `
<DateTimePickerContainer>
  <t
    allowSameDay={false}
    dateFormat="do MMM yyyy HH:mm"
    dateFormatCalendar="LLLL yyyy"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    inlineFocusSelectedMonth={false}
    maxDate={1996-01-01T12:45:00.000Z}
    minDate={1995-12-11T12:45:00.000Z}
    monthsShown={1}
    nextMonthButtonLabel="Next Month"
    nextYearButtonLabel="Next Year"
    onBlur={[Function]}
    onCalendarClose={[Function]}
    onCalendarOpen={[Function]}
    onChange={[MockFunction]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onInputClick={[Function]}
    onInputError={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    preventOpenOnFocus={false}
    previousMonthButtonLabel="Previous Month"
    previousYearButtonLabel="Previous Year"
    readOnly={false}
    renderDayContents={[Function]}
    selected={1995-12-12T12:45:00.000Z}
    shouldCloseOnSelect={true}
    showMonthYearPicker={false}
    showPopperArrow={true}
    showPreviousMonths={false}
    showQuarterYearPicker={false}
    showTimeInput={false}
    showTimeSelect={true}
    showYearDropdown={true}
    strictParsing={false}
    timeCaption="time"
    timeFormat="HH:mm"
    timeInputLabel="Time"
    timeIntervals={15}
    withPortal={false}
  />
</DateTimePickerContainer>
`;
